consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/IbftRound.java

```diff
@@ -86,72 +86,53 @@ public void createAndSendProposalMessage(final long headerTimeStampSeconds) {
 
-    Proposal proposal;
+    final RoundChangeCertificate roundChangeCertificate =
+        roundChangeArtifacts.getRoundChangeCertificate();
+    Block blockToPublish;
     if (!bestBlockFromRoundChange.isPresent()) {
-      LOG.debug("Multicasting NewRound with new block. round={}", roundState.getRoundIdentifier());
-      final Block block = blockCreator.createBlock(headerTimestamp);
-      proposal = messageFactory.createProposal(getRoundIdentifier(), block);
+      LOG.debug("Multicasting Proposal with new block. round={}", roundState.getRoundIdentifier());
+      blockToPublish = blockCreator.createBlock(headerTimestamp);
     } else {
       LOG.debug(
-          "Multicasting NewRound from PreparedCertificate. round={}",
+          "Multicasting Proposal from PreparedCertificate. round={}",
           roundState.getRoundIdentifier());
-      proposal = createProposalAroundBlock(bestBlockFromRoundChange.get());
+      blockToPublish =
+          IbftBlockInterface.replaceRoundInBlock(
+              bestBlockFromRoundChange.get(),
+              getRoundIdentifier().getRoundNumber(),
+              IbftBlockHashing::calculateDataHashForCommittedSeal);
     }
-    transmitter.multicastNewRound(
-        getRoundIdentifier(),
-        roundChangeArtifacts.getRoundChangeCertificate(),
-        proposal.getSignedPayload(),
-        proposal.getBlock());
-    updateStateWithProposedBlock(proposal);
-  }
 
-  private Proposal createProposalAroundBlock(final Block block) {
-    final Block blockToPublish =
-        IbftBlockInterface.replaceRoundInBlock(
-            block,
-            getRoundIdentifier().getRoundNumber(),
-            IbftBlockHashing::calculateDataHashForCommittedSeal);
-    return messageFactory.createProposal(getRoundIdentifier(), blockToPublish);
+    updateStateWithProposalAndTransmit(blockToPublish, Optional.of(roundChangeCertificate));
   }
 
-  public void handleProposalMessage(final Proposal msg) {
-    LOG.debug("Handling a Proposal message.");
+  private void updateStateWithProposalAndTransmit(
+      final Block block, final Optional<RoundChangeCertificate> roundChangeCertificate) {
+    final Proposal proposal =
+        messageFactory.createProposal(getRoundIdentifier(), block, roundChangeCertificate);
 
-    if (getRoundIdentifier().getRoundNumber() != 0) {
-      LOG.error("Illegally received a Proposal message when not in Round 0.");
-      return;
-    }
-    actionReceivedProposal(msg);
-  }
-
-  public void handleProposalFromNewRound(final NewRound msg) {
-    LOG.debug("Handling a New Round Proposal.");
-
-    if (getRoundIdentifier().getRoundNumber() == 0) {
-      LOG.error("Illegally received a NewRound message when in Round 0.");
-      return;
-    }
-    actionReceivedProposal(new Proposal(msg.getProposalPayload(), msg.getBlock()));
+    transmitter.multicastProposal(
+        proposal.getRoundIdentifier(), proposal.getBlock(), proposal.getRoundChangeCertificate());
```

Comment:
Is there any benefit in passing the 3 args as seperate args, could just pass in the proposal?

Reply:
The transmitter was meant to abstract the actually message shape from the client, HOWEVER, it turns out the client needs the actual message such that it can be buffered for use in .. this idea needs rework :/

Reply:
happy with that to be another PR/ticket